spark/src/main/java/org/apache/iceberg/spark/SparkDataFile.java

```diff
@@ -74,42 +105,44 @@ public StructLike partition() {
 
   @Override
   public long recordCount() {
-    return wrapped.getAs(fieldPositions[3]);
+    return wrapped.getAs(recordCountPosition);
   }
 
   @Override
   public long fileSizeInBytes() {
-    return wrapped.getAs(fieldPositions[4]);
+    return wrapped.getAs(fileSizeInBytesPosition);
   }
 
   @Override
   public Map<Integer, Long> columnSizes() {
-    return wrapped.getJavaMap(fieldPositions[6]);
+    return !wrapped.isNullAt(columnSizesPosition) ? wrapped.getJavaMap(columnSizesPosition) : null;
```

Comment:
Unfortunately, Spark does not handle nulls in `getJavaList` and `getJavaMap` correctly. Internally, it calls `.asJava` on null that causes a NPE. I hit that for cases when offsets are not present.